build/media-suite_helper.sh

```diff
@@ -981,7 +981,9 @@ do_patch() {
     local strip=${3:-1}  # value of "patch" -p i.e. leading directories to strip
     [[ $patch = ${patch##*/} ]] &&
         patch="/patches/$patch"
-    do_wget -c -r -q "$patch"
+    if ! ( cp -f "/trunk$patch" $(pwd)>/dev/null 2>&1 ); then    
```

Comment:
Why parenthesis?
Why `$(pwd)` instead of `./`?
Why in `/trunk` instead of inside `/build`?

Reply:
Why should the folder called something else? First of all "official patches" go in there, last is anything else. Exactly in that order. 

Second: there's no option needed. If patch cannot be copied it will wget it.

Reply:
>Why in /trunk instead of inside /build?
Because I've tried both and wasted a lot of time with /build. PLEASE outside!

Reply:
>Why $(pwd) instead of ./?

What's wrong with $(pwd)? Can you please remove your own $(pwd) usage in the script(s) first?

Reply:
>Why parenthesis?
Why not???

Reply:
> Why should the folder called be something else? First of all "official patches" go in there, last is anything else. Exactly in that order.

The "official patches" aren't there locally. They're always downloaded so they're always updated. The alternative would be keep a "patches" git dir/repo inside `build` instead.

> Because I've tried both and wasted a lot of time with /build. PLEASE outside!

What do you mean by wasting time? You could just do `/build${patches}` like you're already doing with `/trunk`. Things needed for build go in /build. There's no direct usage of `/trunk` anywhere except in this PR.

> What's wrong with $(pwd)? Can you please remove your own $(pwd) usage in the script(s) first?
> > Why parenthesis?
> Why not???

Using `$()` and `()` open a sub-shell. In your `cp` case there's absolutely no reason for that. `cp` already returns 0 or non-zero for fails. And I use `$(pwd)` when I need the absolute path, afaik. `cp` doesn't need absolute paths.

If I were to completely change your PR there wouldn't be much point to accepting it and your authorship. I still disagree with making this available since absolutely no one asked for it.

You could overwrite do_patch() inside `$HOME/custom_build_options` too, or any other function you feel isn't suiting to your usecase.